app/views/RoomView/RightButtons.tsx

```diff
@@ -187,34 +187,38 @@ class RightButtonsContainer extends Component<IRightButtonsProps, IRigthButtonsS
 
 	returnLivechat = () => {
 		const { rid } = this.props;
-		showConfirmationAlert({
-			message: i18n.t('Would_you_like_to_return_the_inquiry'),
-			confirmationText: i18n.t('Yes'),
-			onPress: async () => {
-				try {
-					await returnLivechat(rid);
-				} catch (e: any) {
-					showErrorAlert(e.reason, i18n.t('Oops'));
+		if (rid) {
+			showConfirmationAlert({
+				message: i18n.t('Would_you_like_to_return_the_inquiry'),
+				confirmationText: i18n.t('Yes'),
+				onPress: async () => {
+					try {
+						await returnLivechat(rid);
+					} catch (e: any) {
+						showErrorAlert(e.reason, i18n.t('Oops'));
+					}
 				}
-			}
-		});
+			});
+		}
 	};
 
 	placeOnHoldLivechat = () => {
 		const { navigation, rid } = this.props;
-		showConfirmationAlert({
-			title: i18n.t('Are_you_sure_question_mark'),
-			message: i18n.t('Would_like_to_place_on_hold'),
-			confirmationText: i18n.t('Yes'),
-			onPress: async () => {
-				try {
-					await onHoldLivechat(rid);
-					navigation.navigate('RoomsListView');
-				} catch (e: any) {
-					showErrorAlert(e.data?.error, i18n.t('Oops'));
+		if (rid) {
+			showConfirmationAlert({
+				title: i18n.t('Are_you_sure_question_mark'),
+				message: i18n.t('Would_like_to_place_on_hold'),
+				confirmationText: i18n.t('Yes'),
+				onPress: async () => {
+					try {
+						await onHoldLivechat(rid);
+						navigation.navigate('RoomsListView');
+					} catch (e: any) {
+						showErrorAlert(e.data?.error, i18n.t('Oops'));
+					}
 				}
-			}
-		});
+			});
```

Comment:
The `if(rid)` was added because the props `rid` was fixed from `rid: string` to `rid?: string`. The same for the next `if(rid)`

Reply:
We have two situations.
At some point the rid may not even exist? If yes, the if is valid.
If not, the type is wrong and the logic too.

Reply:
It's the first situation.